packages/cli/src/lib/decorators/preconditions/preconditions.spec.ts

```diff
@@ -0,0 +1,92 @@
+import {Preconditions} from './preconditions';
+import type Command from '@oclif/command';
+
+describe('preconditions', () => {
+  const preconditions = new Array<jest.Mock<Boolean>>(5);
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('if all preconditions succeed', () => {
+    it('should executes preconditions the original command', async () => {
+      let counter = 0;
+      const orderChecker = jest.fn();
```

Comment:
> I see you put quite a lot of emphasis in the tests on "the order is respected".

> Is that really an important/meaningful thing to test ?

> I guess it can be, if we have preconditions that depends on one another, but right now it does not seem to be the case that we have such a need.

>The most important thing to verify is that the original function is only called after all preconditions had a chance to execute.

> Maybe we could simplify the test to only focus on this type of verification?
@olamothe , https://github.com/coveo/cli/pull/121#discussion_r604172048

At the first I thought about not caring for the order (I mean, I could have do a `Promise.all(preconditions.map(precondition(target))` instead of the foreach that'd be prettier and neatier).

But then, as you said, I guess it is not crazy to think that we might want to have preconditions depending on one another. So I decided to care about the order and to enforce it. This goes by the test as well.

I think we should not simplify the test if it means they describe less accurately what the function is doing.

I know the unit test may seem daunting here or even overkill, however, I decided to do so because it was in my opinion the best balance between cost and risk:
 - Risk is very high IMO, because this function is bound to be ubiquitous in the code.
 - To cover such risk, we need to have a high confidence that the function is working properly. To that effect, the most efficient way was to unit test it the best I could.
 
 However, if you think the unit test itself is unclear, it may mean I need to reorganize the code and split it better so its more meaningful. If you do think so, could you give me some pointers?